### ISSUE TITLE
Optimize job manager

### ISSUE THREAD

**Issue by Sophie-Xie** [vesoft-inc/nebula#4077]
1.separation by space
2. high and low priority
3. concurrency control (balance leader will be blocked by previous tasks)
4.support task failure errorcode/reason
5.We need an interface to check if a specific type of job is running in job manager. Just like checkIndexJobRunning now, but more general.

**Comment by Sophie-Xie** [comment]
https://github.com/vesoft-inc/nebula/pull/3976
https://github.com/vesoft-inc/nebula/pull/4001
https://github.com/vesoft-inc/nebula/pull/4045
https://github.com/vesoft-inc/nebula/pull/4067
